src/Microsoft.Azure.SignalR/EndpointProvider/ServiceEndpointManager.cs

```diff
@@ -48,32 +48,39 @@ public override IServiceEndpointProvider GetEndpointProvider(ServiceEndpoint end
             return new ServiceEndpointProvider(endpoint, _options);
         }
 
-        private async void OnChange(ServiceOptions options)
+        private void OnChange(ServiceOptions options)
         {
             Log.DetectConfigurationChanges(_logger);
 
+            // synchronize scale for quick and clean status sync
+            ReloadServiceEndpointsAsync(options.Endpoints).Wait();
```

Comment:
We shouldn't be blocking here...

Reply:
We'd complete reload endpoints before accept next change. Most of the operations are not able to easy rollback. Add a task to avoid Wait()?